src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -33,31 +40,53 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
 
             this.extensionHost = host;
             this.solutionTracker = activeSolutionTracker;
+            this.solutionBindingInformationProvider = this.extensionHost.GetService<ISolutionBindingInformationProvider>();
+            this.solutionBindingInformationProvider.AssertLocalServiceIsNotNull();
 
-            this.extensionHost.VisualStateManager.BindingStateChanged += this.SolutionTracker_ActiveSolutionChanged;
+            // the user changed the binding through the Team Explorer
+            this.extensionHost.VisualStateManager.BindingStateChanged += this.VisualStateManager_BindingStateChanged;
+
+            //the solution changed inside the IDE
             this.solutionTracker.ActiveSolutionChanged += this.SolutionTracker_ActiveSolutionChanged;
-            this.CalculateSolutionBinding();
+
+            this.isSolutionPreviouslyBound = this.IsActiveSolutionBound;
+        }
+
+        private void VisualStateManager_BindingStateChanged(object sender, EventArgs e)
+        {
+            this.HandleSolutionBindingChanges();
         }
 
         private void SolutionTracker_ActiveSolutionChanged(object sender, EventArgs e)
         {
-            this.CalculateSolutionBinding();
+            this.HandleSolutionBindingChanges();
         }
 
-        private void CalculateSolutionBinding()
+        private void HandleSolutionBindingChanges()
         {
-            ISolutionBindingSerializer solutionBinding = this.extensionHost.GetService<ISolutionBindingSerializer>();
-            solutionBinding.AssertLocalServiceIsNotNull();
+            if (this.isSolutionPreviouslyBound != this.IsActiveSolutionBound)
+            {
+                this.isSolutionPreviouslyBound = this.IsActiveSolutionBound;
+                this.OnAnalyzersChanged(isBound: this.IsActiveSolutionBound);
+            }
+        }
 
-            BoundSonarQubeProject bindingInfo = solutionBinding.ReadSolutionBinding();
-            this.IsActiveSolutionBound = bindingInfo != null;
+        private void OnAnalyzersChanged(bool isBound)
+        {
+            this.SolutionBindingChanged?.Invoke(this, isBound);
         }
 
-        public bool IsActiveSolutionBound { get; private set; } = false;
+        public bool IsActiveSolutionBound
+        {
+            get
+            {
+                return this.solutionBindingInformationProvider.GetBoundProjects().Any();
```

Comment:
IsActiveSolutionBound needs to return this.solutionBindingInformationProvider.IsSolutionBound() which might explain the perf issue you saw.